Initial styling

lederer opened the pull request:
## Overview

Setup Sass. This lays a foundation for styling the rest of the app.

Style auth page. 
Style global header. 
Style search page.


### Demo

![image](https://user-images.githubusercontent.com/128699/53755005-7967e280-3e83-11e9-95de-a54b7f29970b.png)

![image](https://user-images.githubusercontent.com/128699/53754882-34dc4700-3e83-11e9-9f1f-ffdc2fa63201.png)

![image](https://user-images.githubusercontent.com/128699/53754986-710fa780-3e83-11e9-9beb-323dd56a4c8d.png)


### Notes

- Renamed `index.css` to `taui.css` for clarity.
- Added `focus-visible` polyfill.
- The fewer Amplify components, the easier it is to style the app. So I stripped out a bunch of Amplify components from the global header that I don't think we needed. Holler if I it turns out we needed those.


## Testing Instructions

 * Load auth screen. Confirm it looks per above.
 * Login. Confirm search screen looks per above.
 * Search for accounts. 
 * Confirm clicking an account list item selects the account and opens map view.
 * Confirm deleting an account deletes the account.

lederer commented on taui/yarn.lock:
```diff
@@ -9,7 +9,7 @@
     "@aws-amplify/cache" "^1.0.22"
     "@aws-amplify/core" "^1.0.22"
-    amazon-cognito-auth-js "~1.2.4"
+    amazon-cognito-auth-js "^1.1.9"
```
Ok. Would `~1.1.9` work, or should it be specifically `~1.2.4`?

flibbertigibbet replied:
`1.2.4` was the last version before the breaking change, and the version listed elsewhere in this file as the locked version.

lederer commented:
> The image path references to src/ used here won't work with the staging or production builds, which have a different directory structure. Reverting to something like background: url(/src/BHAlogo.png) no-repeat; in the styles will compile the image into the bundle and fix the issue.

Some follow-up questions:

1) Is this the case even though all of the `configurations/**/setting.yml` files have been updated to include the `img/` folder:

```
entries:
  - src/img/BHAlogo.png:assets/BHAlogo.png
...
```
If so, I can revert it back so the image is located directly in `/src/` again for now. 

2) I also added an svg to `src/img/` (though neglected to put it in `settings.yml` now that I think about it). Should that also live directly in `src/`?

3) Is it necessary to render it via CSS `background` instead as an `img` element? (Would prefer `img`, but ultimately whatever works)

flibbertigibbet commented:
No, paths to `src/` still would still not resolve because there is no `src/` path in production; `assets/` is the top-level path. However, the production paths also work in development. Referencing as `assets/...` instead works both locally and in production: http://localhost:9966/assets/BHAlogo.png.

Here's the logo on staging: https://echolocator.azavea.com/assets/BHAlogo.png.

References to `src/...` only happen to work because the relative files are present.

`url` references in styles get inlined as base64 by the build tool, but `img` `src` references are left as they are. Image files referenced only by `url` wouldn't need to be listed in the configuration files at all, since they get bundled instead into `index.js` with everything else. Anything that should be referenced by path, as with an `img` `src`, should go in the configuration files. So the trash can image should go in the configuration files as well, if it's not going to get inlined as a `url` reference in styles.

lederer commented:
Thanks. That all makes sense. I've updated the URLs to point at the `assets` folder and updated `**/settings.yml` files to include the trash icon (I'll probably revisit the icon strategy later, but this works for now). 

The files are being copied over to `assets` during build. But the browser can't load them. Eg http://localhost:9966/assets/BHAlogo.png doesn't load for me. 

Oddly it doesn't load on staging either: https://echolocator.azavea.com/assets/BHAlogo.png.

flibbertigibbet commented:
The `png` in `assets` has been wrapped in a CommonJS module, with the expectation it will be imported in JS or styles. Copying over images without modification will require changing the build tool. (Browserify wants to wrap everything in modules.) So, I'll have to make that change in `mastarm`.
